pkg/podsecurityadmission/podsecurityadmission.go

```diff
@@ -27,32 +36,53 @@ func GetAdmissionConfigFromCluster(cluster *apisv3.Cluster) *apiserverv1.Admissi
 }
 
 // GetPluginConfigFromTemplate generates a PluginConfig for PodSecurity from a PodSecurityAdmissionConfigurationTemplate
-func GetPluginConfigFromTemplate(template *apisv3.PodSecurityAdmissionConfigurationTemplate) (apiserverv1.AdmissionPluginConfiguration, error) {
+func GetPluginConfigFromTemplate(template *apisv3.PodSecurityAdmissionConfigurationTemplate, k8sVersion string) (apiserverv1.AdmissionPluginConfiguration, error) {
 	plugin := apiserverv1.AdmissionPluginConfiguration{
 		Name: "PodSecurity",
 		Configuration: &runtime.Unknown{
 			ContentType: "application/json",
 		},
 	}
-	psaConfig := psav1.PodSecurityConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: psav1.SchemeGroupVersion.String(),
-			Kind:       "PodSecurityConfiguration",
-		},
+	parsedVersion, err := GetClusterVersion(k8sVersion)
+	if err != nil {
+		return plugin, fmt.Errorf("failed to parse cluster version: %w", err)
 	}
-
-	// here we use JSON to convert the Configuration under template into an instance of PodSecurityConfiguration
+	var cBytes []byte
+	// in the following, we use JSON to convert the Configuration under template into an instance of PodSecurityConfiguration
 	// it works because those two structs have the same JSON tags
 	data, err := json.Marshal(template.Configuration)
 	if err != nil {
 		return plugin, fmt.Errorf("failed to marshal configuration from template: %w", err)
 	}
-	if err = json.Unmarshal(data, &psaConfig); err != nil {
-		return plugin, fmt.Errorf("failed to unmarshal data into PodSecurityConfiguration: %w", err)
-	}
-	cBytes, err := json.Marshal(psaConfig)
-	if err != nil {
-		return plugin, fmt.Errorf("failed to marshal PodSecurityConfiguration: %w", err)
+	switch {
+	case parsedRangeAtLeast125(parsedVersion):
+		psaConfig := psav1.PodSecurityConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: psav1.SchemeGroupVersion.String(),
+				Kind:       "PodSecurityConfiguration",
+			},
+		}
+		if err = json.Unmarshal(data, &psaConfig); err != nil {
+			return plugin, fmt.Errorf("failed to unmarshal data into PodSecurityConfiguration: %w", err)
+		}
+		cBytes, err = json.Marshal(psaConfig)
+		if err != nil {
+			return plugin, fmt.Errorf("failed to marshal PodSecurityConfiguration: %w", err)
+		}
+	case parsedRange123to124(parsedVersion):
+		psaConfig := psav1beta1.PodSecurityConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: psav1beta1.SchemeGroupVersion.String(),
+				Kind:       "PodSecurityConfiguration",
+			},
+		}
+		if err = json.Unmarshal(data, &psaConfig); err != nil {
+			return plugin, fmt.Errorf("failed to unmarshal data into PodSecurityConfiguration: %w", err)
+		}
+		cBytes, err = json.Marshal(psaConfig)
+		if err != nil {
+			return plugin, fmt.Errorf("failed to marshal PodSecurityConfiguration: %w", err)
+		}
```

Comment:
The marshaling code is the same in both cases and can be moved out of the switch with something like this.
```suggestion
	var psaConfig runtime.Object
	switch {
	case parsedRangeAtLeast125(parsedVersion):
		psaConfig = &psav1.PodSecurityConfiguration{
			TypeMeta: metav1.TypeMeta{
				APIVersion: psav1.SchemeGroupVersion.String(),
				Kind:       "PodSecurityConfiguration",
			},
		}
	case parsedRange123to124(parsedVersion):
		psaConfig = &psav1beta1.PodSecurityConfiguration{
			TypeMeta: metav1.TypeMeta{
				APIVersion: psav1beta1.SchemeGroupVersion.String(),
				Kind:       "PodSecurityConfiguration",
			},
		}
	}
	if err = json.Unmarshal(data, psaConfig); err != nil {
		return plugin, fmt.Errorf("failed to unmarshal data into PodSecurityConfiguration: %w", err)
	}
	cBytes, err = json.Marshal(psaConfig)
	if err != nil {
		return plugin, fmt.Errorf("failed to marshal PodSecurityConfiguration: %w", err)
	}
```